snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,383 @@
+    return uid
+
+
+def _get_jhbuild_group():
+    mygid = os.getgid()
+    myuid = os.getuid()
+    if myuid != 0:
+        return mygid
+
+    try:
+        gid = common.run_output(['id', '-g', 'jhbuild'])
+    except subprocess.CalledProcessError:
+        gid = 65534
+    return gid
+
+
+class JHBuildPlugin(snapcraft.BasePlugin):
+    """
+    JHBuildPlugin provides jhbuild functionality to snapcraft
+    """
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+
+        schema['properties'] = {
+            'modules': {
+                'type': 'array',
+                'items': {
+                    'type': 'string',
+                },
+                'minItems': 1,
+                'uniqueItems': True,
+            },
+            'module-set': {
+                'type': 'string',
+                'default': 'gnome-world',
+            },
+            'module-set-dir': {
+                'type': 'string',
+                'default': '',
+            },
+            'jhbuild-archive': {
+                'type': 'string',
+                'default': '',
+            },
+            'jhbuild-mirror': {
+                'type': 'string',
+                'default': '',
+            },
+            'cflags': {
+                'type': 'string',
+                'default': '',
+            },
+        }
+
+        schema['required'] = ['modules']
+
+        return schema
+
+    @classmethod
+    def get_pull_properties(cls):
+        # Inform Snapcraft of the properties associated with pulling. If these
+        # change in the YAML Snapcraft will consider the build step dirty.
+        return [
+            'modules',
+            'module-set',
+            'module-set-dir',
+            'jhbuild-archive',
+            'jhbuild-mirror',
+        ]
+
+    @classmethod
+    def get_build_properties(cls):
+        # Inform Snapcraft of the properties associated with building. If these
+        # change in the YAML Snapcraft will consider the build step dirty.
+        return [
+            'cflags',
+        ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.modules = [
+            module for module in self.options.modules if module[0] != '-']
+
+        self.skip_modules = [
+            module[1:] for module in self.options.modules if module[0] == '-']
+
+        self.build_packages += list(BUILD_PACKAGES)
+
+        self.jhbuild_src = os.path.join(self.partdir, 'jhbuild', 'src')
+        self.jhbuild_program = os.path.join(
+            self.partdir, 'jhbuild', 'usr', 'bin', 'jhbuild')
+        self.jhbuildrc_path = os.path.join(self.partdir, 'jhbuildrc')
+
+    def enable_cross_compilation(self):
+        """
+        We don't support cross compilation (yet?)
+        """
+        raise NotImplementedError(
+            'The {!s} plugin does not support cross compilation'.format(
+                self.name))
+
+    def run(self, cmd, cwd=None, **kwargs):
```

Comment:
Why are you defining run and run_output only to call the definitions from the supper class? It would be the same to just remove them from this class.